### ISSUE TITLE
about batch_norm_manual

### ISSUE THREAD

**Issue by SunskyF** [ptrblck/pytorch_misc#5]
[Here](https://github.com/ptrblck/pytorch_misc/blob/20e8ea93bd458b88f921a87e2d4001a4eb753a02/batch_norm_manual.py#L63) use the external `x`, seems not good

**Comment by ptrblck** [comment]
@SunskyF Thanks for pointing this out! I'll push a fix to the repo.

**Comment by LiYingwei** [comment]
It seems that when I use multiple GPU (via DataParallel) for training, the behavior is not the same as nn.BatchNorm2d. Do you have any hint about this?

**Comment by jinensetpal** [comment]
This response is very late, but batch statistics are not synced by default, which might be why the behavior is different. https://pytorch.org/docs/stable/generated/torch.nn.SyncBatchNorm.html corrects this by syncing statistics across each device.
